doc: releases: Intermediate update for new 4.2 boards/samples/drivers

kartben opened the pull request:
New boards, drivers, samples, and shields since June 10, 2025.

kartben committed:
doc: releases: Intermediate update for new 4.2 boards/samples/drivers

New boards, drivers, samples, and shields since June 10, 2025

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

danieldegrasse commented on doc/releases/release-notes-4.2.rst:
```diff
@@ -684,22 +750,28 @@ New Drivers
    * :dtcompatible:`raspberrypi,csi-22pins-connector`
    * :dtcompatible:`raspberrypi,pico-gpio-port`
    * :dtcompatible:`renesas,ra-parallel-graphics-header`
-   * :dtcompatible:`renesas,rx-gpio`
```
Was dropping this intentional?

kartben committed:
dts: bindings: update binding-types.txt with recently added types

SENT, PSI5, Virtio where all recently added types. Add them to the
binding-types.txt file so that their full name can be displayed in the
documentation.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

danieldegrasse commented:
@dkalowsk PTAL
